[Lee Yi Sheng] iP

Lee-YiSheng opened the pull request.

Lee-YiSheng committed:
Level 0. Rename, Greet, Exit

Refactored chatbot name to Pep and implemented skeletal version

Lee-YiSheng committed:
Added Echo functionality

Program now waits for user input, will echo the input unless receives "bye", where it then exists.

Lee-YiSheng committed:
Encapsulated UI and methods

All UI logic is in Ui class, and improved readability with Pep focusing on chatbot behavior while UI handles presentation.

Lee-YiSheng committed:
Add, List

Add the ability to store whatever text entered by the user and display them back to the user when requested.

Lee-YiSheng committed:
Mark as Done

Add the ability to mark tasks as done. Optionally, add the ability to change the status back to not done.

Lee-YiSheng committed:
refactor: split Pep into separate classes

- Moved Ui, TaskList, and Task into their own files for better modularity

Lee-YiSheng committed:
refactor: improve task output formatting in toString()

Replaced direct System.out.println usage with a cleaner toString() override
that formats tasks consistently using status icon and description.

Lee-YiSheng committed:
feat: add support for Todo, Deadline task types

- Implemented Task subclasses (Todo, Deadline) inheriting from Task
- Updated Pep to parse commands for each task type and store them polymorphically
- Enhanced UI to display task type, status, and relevant date details

Lee-YiSheng committed:
feat: add Event task type with start and end time support

- Implemented Event class extending Task to represent tasks with a start and end time
- Updated command parsing in Pep to handle 'event' input with /from and /to parameters
- Enhanced UI to display Event tasks with their time range

sandrej09 commented on src/main/java/Pep.java:
```diff
@@ -1,11 +1,23 @@
         String chatbotName = "Pep";
-        System.out.println("____________________________________________________________");
+        String line = "____________________________________________________________";
+        System.out.println(line);
```
The underscores are cool, but maybe you don’t need to repeat them so many times. Could be easier if it was just one variable you reuse.

sandrej09 commented on src/main/java/Pep.java:
```diff
@@ -1,11 +1,23 @@
+        System.out.println(line);
+        Scanner scanner = new Scanner(System.in);  // Create a Scanner object
+        String userInput =  scanner.nextLine();
+        while (!(userInput.equals("bye"))) {
```
The first nextLine before the loop feels a bit unnecessary. Would it be clearer to use a do { ...}   while (...) loop instead, so you don’t duplicate nextLine() and the program shows something first before reading again?

sandrej09 commented on src/main/java/Pep.java:
```diff
@@ -29,14 +29,18 @@ public void run() {
-                int index = parseIndex(userInput);
-                if (index != -1 && taskList.unmarkTask(index)) {
-                    ui.showMarked(taskList.getTask(index));
+                String[] parts = userInput.substring(9).split(" /by ", 2);
```
What happens if the user forgets to type /by for a deadline? Maybe you could add some feedback message for that case.

sandrej09 replied:
Variable naming like parts could be more descriptive  --  (descriptionAndDate), might make it easier for others to understand quickly.

ycdaniel326 commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,57 @@
+public class Ui {
```
I like the way that you created a Ui Class for displaying. It makes the code of the main method much shorter and readable.

ycdaniel326 commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,40 @@
+            count++;
+        }
+    }
+
```
It is great that you added the checking conditions for each command. It can ensure the program runs smoothly. 
Perhaps you can add some lines to indicate that the action is not feasible, e.g. "TaskList is full!"

Lee-YiSheng committed:
update code according to instructions

Lee-YiSheng committed:
update expected.txt for text-ui-test output changes

Synced expected output for welcome message

Lee-YiSheng committed:
add Command abstract class

Lee-YiSheng committed:
add ListCommand class

Lee-YiSheng committed:
add MarkCommand class

Lee-YiSheng committed:
add UnmarkCommand class

Lee-YiSheng committed:
add AddTodoCommand class

Lee-YiSheng committed:
add ExitCommand class

Lee-YiSheng committed:
add Parser class

Lee-YiSheng committed:
Updated parser class and refactored Pep's run()

Parser now includes todo, deadline and event commands.

Lee-YiSheng committed:
add AddDeadlineCommand class

Lee-YiSheng committed:
add AddEventCommand class

Lee-YiSheng committed:
Refactor: Packaged commands

commands have been packaged and code refactored to include relevant imports

Lee-YiSheng committed:
Refactor: Packaged tasks

tasks, deadline, event, todo have been packaged

Lee-YiSheng committed:
README.md updated for autocomplete acknowledgement

Lee-YiSheng committed:
refactor: Packaged ui and refactored TaskList to package task

Lee-YiSheng committed:
bug fix for AddTodoCommand

AddTodoCommand class had a mistyped object of Task instead of Todo

Lee-YiSheng committed:
Handle Errors

added IllegalArgumentException to show error message when user input unusual commands

Lee-YiSheng committed:
add exceptions package, PepException class

Lee-YiSheng committed:
Added exception handling for todo

Lee-YiSheng committed:
Added exception handling for AddEventCommand

Lee-YiSheng committed:
Updated exception handling for AddEventCommand to handle missing 'from' and 'to'

Lee-YiSheng committed:
merge: Level-5 increment

roshnidaksha commented on src/main/java/Pep.java:
```diff
@@ -0,0 +1,40 @@
+                Command command = Parser.parse(userInput);
+                command.execute(taskList, ui);
+                isExit = command.isExit();
+            } catch (Exception e) {
```
Catching `Exception` is too generic. Consider adding some custom/ specific exceptions.

roshnidaksha commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,42 @@
+        }
+    }
+
+    public boolean markTask(int index) {
```
Perhaps using a better prefix would make this method sound like Boolean. The same applies to method `unmarkTask` as well.

roshnidaksha commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,27 @@
+import commands.*;
+
+public class Parser {
```
`CommandParser` could be a better name

roshnidaksha commented on src/main/java/ui/Ui.java:
```diff
@@ -0,0 +1,66 @@
+        System.out.println(" added: " + task);
+        System.out.println(" Now you have " + totalTasks + " tasks in the list.");
+        printLine();
+
+
```
Avoid leaving unnecessary blank lines.

roshnidaksha reviewed (commented):
Overall, well written code. 
There are some minor things which I have mentioned in the comments.

roshnidaksha commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,27 @@
+import commands.*;
```
Imported classes should always be listed explicitly. Avoid using wild card import statements.

Lee-YiSheng committed:
Refactor: TaskList to use ArrayList for dynamic task

Lee-YiSheng committed:
refactor: update markTask and unmarkTask to throw PepException

Replace boolean return values with PepException for invalid task indices.
Update MarkCommand and UnmarkCommand to declare throws PepException.

Lee-YiSheng committed:
feat: add DeleteCommand to remove tasks by index

Implements DeleteCommand to handle task deletion from TaskList.
Validates index bounds and throws PepException for invalid task numbers.

Lee-YiSheng committed:
feat: add DeleteCommand to remove tasks by index

Updates UI to display confirmation of deleted task and remaining count.

Lee-YiSheng committed:
add Storage class

Storage class for saving files

Lee-YiSheng committed:
feat: implement persistent storage for task list

Add Storage class to handle saving and loading tasks from disk using a
relative, OS-independent file path (./data/duke.txt). Automatically
create the data directory and file if they do not exist. Save the task
list after every modifying command (add, delete, mark, unmark) and load
existing tasks on startup. Include error handling for missing files and
corrupted data lines to ensure robustness across different environments.

Lee-YiSheng committed:
fix(storage): prevent task list from clearing on startup

Rewrite Storage.load() to correctly load tasks from existing file without
overwriting when the file is present. Add debug output to confirm file
path, skip corrupted or blank lines instead of aborting, and only create
data directory and file if missing. This ensures saved tasks persist
across chatbot restarts.

Lee-YiSheng committed:
merge: Level-6 increment

Lee-YiSheng committed:
merge: Level-7 increment

# Conflicts:
#	src/main/java/tasks/TaskList.java

Lee-YiSheng committed:
feat(tasklist): restore deleteTask method to remove tasks by index

Reintroduce deleteTask(int index) to TaskList, allowing removal of a task
at a given index. Method marks the task as not done before removal and
returns the removed Task. This restores previously removed functionality
needed by DeleteCommand and related features.

Lee-YiSheng committed:
feat(parser): add support for delete command

Update Parser to recognize 'delete <task number>' input and return a
DeleteCommand with the correct zero-based index. This enables users to
remove tasks from the list via the delete command, consistent with mark
and unmark parsing logic.

Lee-YiSheng committed:
refactor: removed empty lines

Lee-YiSheng committed:
feat(find): implement keyword search for tasks

Add findTasks method in TaskList to return all tasks containing a given
keyword. Update FindCommand to use this method and Ui.showTasks to
display matching tasks with numbering. Supports multiple matches and
handles empty results gracefully.

Lee-YiSheng committed:
refactor: removed boolean return of markTask

Previously markTask return true or false as an error checker, now redundant due to exception handling.

Lee-YiSheng committed:
docs: add JavaDoc comments to core classes and methods

Add JavaDoc headers to Parser, FindCommand, TaskList, Storage, and Ui.
Document key public methods including parse, execute, deleteTask,
findTasks, load, save, and UI display methods. Improves readability and
maintainability by clarifying the purpose, parameters, and behavior of
major components.

Lee-YiSheng committed:
refactor: clearer parameter names for chatbot

welcome message was using generic 'name' as variable, refactored to chatbotName for clearer usage.

Lee-YiSheng committed:
chore(storage): remove redundant null check and handle mkdirs/createNewFile results

Eliminate unnecessary 'task != null' check in Storage.load since parseTask
never returns null. Add explicit handling of File.mkdirs() and
File.createNewFile() return values to acknowledge success or failure and
suppress IDE warnings. Improves code clarity and robustness.

Lee-YiSheng committed:
docs: add user guide for Pep chatbot

Create docs/README.md with a concise user guide. Includes quick start
instructions, feature descriptions with examples, summary table of
commands, and FAQ. Provides enough detail for new users to use all
important features while keeping the guide short and user-friendly.

Lee-YiSheng committed:
Merge branch 'branch-Level-9'

Lee-YiSheng committed:
deleted README.md

Lee-YiSheng committed:
refactor(parser): centralize numeric argument parsing

Introduce parseIndex helper in Parser to handle integer argument parsing
for mark, unmark, and delete commands. Ensures consistent error handling,
usage messages, and prevents crashes on invalid input. Removes duplicate
try/catch logic across commands.

Lee-YiSheng committed:
refactor(ui): improve error display and integrate with index validation

Update Ui.showError to display PepException messages in a consistent
boxed format. Refactor MarkCommand, UnmarkCommand, and DeleteCommand to
use TaskList.validateIndex for bounds checking, ensuring invalid task
numbers are handled gracefully without crashing the program.

Lee-YiSheng committed:
docs(readme): fix invalid YAML front matter in User Guide

Remove incorrect '---' markers from docs/README.md that caused Jekyll
build to fail. The file now begins directly with Markdown content
(`# Pep Chatbot – User Guide`). Ensures compatibility with GitHub Pages
and prevents invalid YAML parsing errors.

Lee-YiSheng committed:
docs(deadline): add JavaDoc for Deadline class

Document the Deadline task with class-level and method-level JavaDoc.
Explain how it stores due dates using LocalDate, the expected input
format (yyyy-MM-dd), and the formatted output (MMM dd yyyy). Improves
code readability and maintainability by clarifying the purpose and
behavior of the Deadline class.

Lee-YiSheng committed:
Merge branch 'master' into branch-A-JavaDoc

Lee-YiSheng committed:
docs(tasks): add JavaDoc and LocalDate/LocalDateTime support for Deadline and Event

Add comprehensive JavaDoc to Deadline and Event classes. Refactor Deadline
to store due dates as LocalDate and Event to store start/end times as
LocalDateTime. Input accepted in ISO-like formats (yyyy-MM-dd, yyyy-MM-dd HHmm)
and displayed in user-friendly formats (e.g., Oct 15 2019, Dec 02 2019 18:00).

Lee-YiSheng committed:
feat(storage): support ISO date/time serialization for Deadline and Event

Update formatTask to save LocalDate and LocalDateTime in ISO format.
Update parseTask to reconstruct Deadline and Event from these values.
Revise User Guide examples for deadline and event commands to use
yyyy-MM-dd and yyyy-MM-dd HHmm input formats, displayed as
MMM dd yyyy and MMM dd yyyy HH:mm.

Lee-YiSheng committed:
docs(userguide): update deadline and event formats to use ISO date/time

Revise User Guide examples for deadline and event commands to accept
yyyy-MM-dd and yyyy-MM-dd HHmm input formats. Update displayed output
examples to show user-friendly formats (MMM dd yyyy, MMM dd yyyy HH:mm).
Ensures documentation matches new LocalDate/LocalDateTime implementation.

Lee-YiSheng committed:
Merge pull request #1 from Lee-YiSheng/feature/dates-and-times

Merging PRs:

Lee-YiSheng committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/tasks/Deadline.java

Lee-YiSheng committed:
refactor: fix deadline.java merge issue

Tidy up merge conflict issue in daedline.java, resolving to the original latest code.
